src/which.js

```diff
@@ -47,20 +51,16 @@ function _which(options, cmd) {
 
       if (common.platform === 'win') {
         var baseAttempt = attempt;
-        attempt = baseAttempt + '.exe';
-        if (checkPath(attempt)) {
-          where = attempt;
-          return;
-        }
-        attempt = baseAttempt + '.bat';
-        if (checkPath(attempt)) {
-          where = attempt;
-          return;
-        }
-        attempt = baseAttempt + '.cmd';
-        if (checkPath(attempt)) {
-          where = attempt;
-          return;
+        var pathExtEnv = process.env.PATHEXT || XP_DEFAULT_PATHEXT;
+        var pathExtArray = splitPath(pathExtEnv);
+        var len = pathExtArray.length;
```

Comment:
I don't know if it's part of the standard or not. Either way is fine with me, since the meaning is very clear (and performance isn't super critical yet).